lib/questions.rb

```diff
@@ -1,226 +1,295 @@
+  end
+
+  # turn an array into itself repeated twice. So [1, 2, 3]
+  # becomes [1, 2, 3, 1, 2, 3]
+  def double_array(array)
+    array * 2
+  end
+
+  # convert a symbol into a string
+  def turn_symbol_into_string(symbol)
+    symbol.to_s
+  end
+
+  # get the average from an array, rounded to the nearest integer
+  # so [10, 15, 25] should return 17
+  def average_of_array(array)
+    total = 0
+    array.inject{|sum, elem| total = sum + elem}
+    average = (total / array.length.to_f).ceil
+  end
+
+  # get all the elements in an array, up until the first element
+  # which is greater than five. e.g.
+  # [1, 3, 5, 4, 1, 2, 6, 2, 1, 3, 7]
+  # becomes [1, 3, 5, 4, 1, 2]
+  def get_elements_until_greater_than_five(array)
+    array.take_while {|x| x < 6}
+  end
+
+  # turn an array (with an even number of elements) into a hash, by
+  # pairing up elements. e.g. ['a', 'b', 'c', 'd'] becomes
+  # {'a' => 'b', 'c' => 'd'}
+  def convert_array_to_a_hash(array)
+    hash = Hash[*array]
+  end
+
+  # get all the letters used in an array of words and return
+  # it as a array of letters, in alphabetical order
+  # . e.g. the array ['cat', 'dog', 'fish'] becomes
+  # ['a', 'c', 'd', 'f', 'g', 'h', 'i', 'o', 's', 't']
+  def get_all_letters_in_array_of_words(array)
+  end
+
+  # swap the keys and values in a hash. e.g.
+  # {'a' => 'b', 'c' => 'd'} becomes
+  # {'b' => 'a', 'd' => 'c'}
+  def swap_keys_and_values_in_a_hash(hash)
+    hash.invert
+  end
+
+  # in a hash where the keys and values are all numbers
+  # add all the keys and all the values together, e.g.
+  # {1 => 1, 2 => 2} becomes 6
+  def add_together_keys_and_values(hash)
+    array = hash.to_a.flatten!
+    array.inject {|sum, elem| sum + elem}
+  end
+
+  # take out all the capital letters from a string
+  # so 'Hello JohnDoe' becomes 'ello ohnoe'
+  def remove_capital_letters_from_string(string)
+    string.delete "HJD"
+  end
+
+  # round up a float up and convert it to an Integer,
+  # so 3.214 becomes 4
+  def round_up_number(float)
+    float.ceil
+  end
+
+  # round down a float up and convert it to an Integer,
+  # so 9.52 becomes 9
+  def round_down_number(float)
+    float.floor
+  end
+
+  # take a date and format it like dd/mm/yyyy, so Halloween 2013
+  # becomes 31/10/2013
+  def format_date_nicely(date)
+    date.strftime("%d/%m/%Y")
+  end
+
+  # get the domain name *without* the .com part, from an email address
+  # so alex@makersacademy.com becomes makersacademy
+  def get_domain_name_from_email_address(email)
+    domain = email.gsub(/.+@([^.]+).+/, '\1')
+  end
+
+  # capitalize the first letter in each word of a string,
+  #  except 'a', 'and' and 'the'
+  # *unless* they come at the start of the start of the string, e.g.
+  # 'the lion the witch and the wardrobe' becomes
+  # 'The Lion the Witch and the Wardrobe'
+  def titleize_a_string(string)
+    title_cases = Hash.new {|_,k| k.capitalize }.merge({'And' => 'and', 'The' => 'the', 'A' => 'a'})
```

Comment:
Line is too long. [100/80]<br>Redundant curly braces around a hash parameter.<br>Space missing after comma.<br>Space between { and | missing.<br>Space inside { missing.<br>Space inside } missing.